### PR TITLE
Create package client and rearrange code

### PR DESCRIPTION
**Pull request by sinhaashish**

### CODE REVIEW

File: cmd/kubectl-direct_csi/utils.go [hunk 1]
```diff
@@ -20,16 +20,18 @@ import (
...
+	"k8s.io/klog/v2"
+
 	"github.com/dustin/go-humanize"
 	"github.com/fatih/color"
 
 	directcsi "github.com/minio/direct-csi/pkg/apis/direct.csi.min.io/v1beta3"
+	"github.com/minio/direct-csi/pkg/client"
 	clientset "github.com/minio/direct-csi/pkg/clientset/typed/direct.csi.min.io/v1beta3"
 	"github.com/minio/direct-csi/pkg/ellipsis"
 	"github.com/minio/direct-csi/pkg/sys"
```
Comment (Praveenrajmani) [inline cmd/kubectl-direct_csi/utils.go:37]: aliased imports can be grouped in order

```
        directcsi "github.com/minio/direct-csi/pkg/apis/direct.csi.min.io/v1beta3"
	clientset "github.com/minio/direct-csi/pkg/clientset/typed/direct.csi.min.io/v1beta3"
        "github.com/minio/direct-csi/pkg/client"
	"github.com/minio/direct-csi/pkg/ellipsis"
	"github.com/minio/direct-csi/pkg/sys"
```

File: pkg/client/colors.go [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+import (
+	"github.com/fatih/color"
+)
+
+// Color print functions.
+var (
+	Bold = color.New(color.Bold).SprintFunc()
+	Red  = color.New(color.FgRed).SprintFunc()
+)
```
Comment (Praveenrajmani) [inline pkg/client/colors.go:27]: These definitions can be in `utils` pkg

File: pkg/client/fake.go [hunk 3]
```diff
@@ -60,7 +61,7 @@ func FakeInit() {
 	_ = metav1.AddMetaToScheme(scheme)
 	metadataClient = metadatafake.NewSimpleMetadataClient(scheme)
 
-	initEvent(kubeClient)
+	utils.InitEvent(kubeClient)
```
Comment (Praveenrajmani) [inline pkg/client/fake.go:64]: `InitEvent` function communicates with k8s. Can we keep them in `client` pkg?

File: pkg/client/list.go [hunk 4]
```diff
@@ -23,12 +23,13 @@ import (
...
 	directcsi "github.com/minio/direct-csi/pkg/apis/direct.csi.min.io/v1beta3"
 	clientset "github.com/minio/direct-csi/pkg/clientset/typed/direct.csi.min.io/v1beta3"
+	"github.com/minio/direct-csi/pkg/utils"
 
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/klog/v2"
 )
 
-func toLabelSelector(labels map[LabelKey][]LabelValue) string {
+func toLabelSelector(labels map[utils.LabelKey][]utils.LabelValue) string {
```
Comment (Praveenrajmani) [inline pkg/client/list.go:32]: this function `toLabelSelector` can be moved to `utils` pkg

File: pkg/utils/event.go [hunk 5]
```diff
@@ -29,7 +29,7 @@ var (
 	eventRecorder    record.EventRecorder
 )
 
-func initEvent(kubeClient kubernetes.Interface) {
+func InitEvent(kubeClient kubernetes.Interface) {
```
Comment (Praveenrajmani) [inline pkg/utils/event.go:32]: This function can be moved to `client` pkg

File: cmd/kubectl-direct_csi/utils.go (hunk 1 above)
Comment (sinhaashish) [reply cmd/kubectl-direct_csi/utils.go]: upon saving the file `"github.com/minio/direct-csi/pkg/client"` got inserted in between `directcsi` and `clientset`

File: pkg/client/event.go [hunk 6]
```diff
@@ -37,7 +37,7 @@ func initEvent(kubeClient kubernetes.Interface) {
 		},
 	)
 	eventRecorder = eventBroadcaster.NewRecorder(
-		Scheme, v1.EventSource{Component: "directcsi-controller"},
+		runtime.NewScheme(), v1.EventSource{Component: "directcsi-controller"},
```
Comment (Praveenrajmani) [inline pkg/client/event.go:40]: We don't need to use `runtime.NewScheme()` here, we can copy `pkg/utils/scheme.go` -> `pkg/client`

**Review by Praveenrajmani — DISMISSED** [review]
LGTM

File: cmd/kubectl-direct_csi/drives_format_test.go [hunk 7]
```diff
@@ -58,7 +59,7 @@ func TestFormatDrivesByAttributes(t1 *testing.T) {
 
 	createTestDrive := func(node, drive, path string, driveStatus directcsi.DriveStatus, accessTier directcsi.AccessTier) *directcsi.DirectCSIDrive {
 		csiDrive := &directcsi.DirectCSIDrive{
-			TypeMeta: utils.DirectCSIDriveTypeMeta(),
+			TypeMeta: client.DirectCSIDriveTypeMeta(),
```
Comment (wlan0) [inline cmd/kubectl-direct_csi/drives_format_test.go:62]: This should be in util

File: cmd/kubectl-direct_csi/utils.go [hunk 8]
```diff
@@ -20,16 +20,19 @@ import (
 	"context"
 	"errors"
 	"fmt"
-	"k8s.io/klog/v2"
 	"os"
 	"regexp"
 	"strings"
 	"time"
 
+	"k8s.io/klog/v2"
```
Comment (wlan0) [inline cmd/kubectl-direct_csi/utils.go:28]: klog should be the last import. This convention is good to follow

File: pkg/client/client.go [hunk 9]
```diff
@@ -14,23 +14,119 @@
...
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/restmapper"
+	"k8s.io/client-go/tools/clientcmd"
+
+	directcsi "github.com/minio/direct-csi/pkg/apis/direct.csi.min.io/v1beta3"
+	"github.com/minio/direct-csi/pkg/utils"
 
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
```
Comment (wlan0) [inline pkg/client/client.go:40]: keep all kubernetes imports together. 
keep all direct-csi imports together. 
keep all other imports together. 
keep core imports at the top. 
keep klog as the last import.

File: pkg/controller/controller.go [hunk 10]
```diff
@@ -295,7 +296,7 @@ func (c *ControllerServer) DeleteVolume(ctx context.Context, req *csi.DeleteVolu
 	vclient := directCSIClient.DirectCSIVolumes()
 
 	vol, err := vclient.Get(ctx, vID, metav1.GetOptions{
-		TypeMeta: utils.DirectCSIVolumeTypeMeta(),
+		TypeMeta: client.DirectCSIVolumeTypeMeta(),
```
Comment (wlan0) [inline pkg/controller/controller.go:299]: this should be in utils

File: pkg/controller/controller.go [hunk 11]
```diff
@@ -326,7 +327,7 @@ func (c *ControllerServer) DeleteVolume(ctx context.Context, req *csi.DeleteVolu
 	vol.SetFinalizers(updatedFinalizers)
 
 	_, err = vclient.Update(ctx, vol, metav1.UpdateOptions{
-		TypeMeta: utils.DirectCSIVolumeTypeMeta(),
+		TypeMeta: client.DirectCSIVolumeTypeMeta(),
```
Comment (wlan0) [inline pkg/controller/controller.go:330]: this should be in utils

**Review by wlan0 — APPROVED** [review]

**Pull request merged**